test/functional/readpreference.test.js

```diff
@@ -548,4 +550,102 @@ describe('ReadPreference', function() {
       client.close(done);
     });
   });
+
+  context('hedge', function() {
+    before(function() {
+      this.testHedge = callback => {
+        const { configuration } = this;
+        const client = configuration.newClient(configuration.writeConcernMax(), {
+          poolSize: 1,
+          monitorCommands: true
+        });
+        const events = [];
+        client.on('commandStarted', filterForCommands(['find'], events));
+        client.connect(function(err, client) {
+          test.equal(null, err);
+          const collection = client.db(configuration.db).collection('test');
+          return callback(client, collection, events);
+        });
+      };
+    });
+
+    const itWithMeta = (a, b) => {
```

Comment:
Given what this is doing, I think a better name for this might be `onlySharded` or `onlyTestOnSharded` or something like that.